test/benchmark_gtest.cc

```diff
@@ -4,6 +4,12 @@
 #include "gmock/gmock.h"
 #include "gtest/gtest.h"
 
+namespace benchmark { namespace internal {
+
+std::string drop_last_field (const std::string &, char);
```

Comment:
That's ... interesting.
Honestly, i'm not sure this will always work.
Can you instead stick the declaration into some header?
`src/complexity.h` perhaps?

Reply:
It _probably_ will work. I'm just conflicted about the fact that the function you want to call
in the test, and the actual implementation of the function don't have the _same_ declaration.
Maybe that is ok/allowed, don't know. Looks like a code smell at least.

Reply:
I'm fairly confident that `const T` and `T` for parameters are interchangeable. I personally mark them `const` in the definition to document that I won't modify the argument in the function body, and not `const` in the header since the caller needn't care for by-value parameters.

All that said, I can't find anything in the standard from a quick glance ensuring that this is OK, and anyway I'm happy to comply with the local conventions.

Reply:
This is perfectly reasonable, especially the notion of dropping the const from value types in API declarations. It's a _little_ odd to see it as an extern declaration but it works perfectly.